src/gpu/intel/jit/dsl/dsl.cpp

```diff
@@ -275,114 +300,159 @@ tensor_t def(const v2::layout_t &layout, const std::string &name,
     return default_ctx().def(layout, name, value);
 }
 
-lval_t &lval_t::operator=(const expr_t &obj) {
-    assign(this->var, obj);
+lval_t &lval_t::operator=(expr_t obj) {
+    assign(this->var, std::move(obj));
     return *this;
 }
 
-expr_t let(type_t type, const std::string &name, const expr_t &value) {
-    return default_ctx().let(type, name, value);
+expr_t let(type_t type, const std::string &name, expr_t value) {
+    return default_ctx().let(type, name, std::move(value));
 }
 
-expr_t let(const std::string &name, const expr_t &value) {
-    return default_ctx().let(name, value);
+expr_t let(const std::string &name, expr_t value) {
+    return default_ctx().let(name, std::move(value));
 }
 
 template <>
-void if_(const expr_t &cond, const stmt_t &if_body) {
+void if_(expr_t cond, stmt_t if_body) {
     if (is_const(cond))
-        default_ctx().append(to_cpp<bool>(cond) ? if_body : stmt_t());
+        append(to_cpp<bool>(cond) ? std::move(if_body) : stmt_t());
     else
-        default_ctx().append(if_t::make(cond, if_body));
+        append(if_t::make(std::move(cond), std::move(if_body)));
 }
+
 template <>
-void if_(const expr_t &cond, const stmt_t &if_body, const stmt_t &else_body) {
+void if_(expr_t cond, stmt_t if_body, stmt_t else_body) {
     if (is_const(cond))
-        default_ctx().append(to_cpp<bool>(cond) ? if_body : else_body);
+        append(to_cpp<bool>(cond) ? std::move(if_body) : std::move(else_body));
     else
-        default_ctx().append(if_t::make(cond, if_body, else_body));
+        append(if_t::make(
+                std::move(cond), std::move(if_body), std::move(else_body)));
 }
 template <>
-void while_(const expr_t &cond, const stmt_t &body) {
-    default_ctx().append(while_t::make(cond, body));
+void while_(expr_t cond, stmt_t body) {
+    append(while_t::make(std::move(cond), std::move(body)));
 }
 
-void assign(const expr_t &var, const expr_t &value) {
-    default_ctx().append(store_t::make(var, 0, value));
+void assign(expr_t var, expr_t value) {
+    append(store_t::make(std::move(var), 0, std::move(value)));
 }
 
 enum class send_kind_t { load, prefetch, store };
 
 void scatter_send(const tensor_t &t, const global_tensor_t &g,
-        const transform_t &transform, send_kind_t &op_kind,
-        const icoord_t &base) {
-    gpu_warning() << "Scatter messages are not yet implemented.";
+        send_kind_t &op_kind, const icoord_t &base, const send_hint_t &hint) {
+    gpu_warning() << "Scatter messages are not yet implemented\n";
```

Comment:
Nit: `\n` is automatic.

```suggestion
    gpu_warning() << "Scatter messages are not yet implemented";
```